Library/Homebrew/extend/os/mac/caveats.rb

```diff
@@ -1,56 +0,0 @@
-# typed: true
```

Comment:
Would prefer the `OS.mac?`, `OS.linux?` logic moved to this file instead.

Reply:
It's not specific to caveats though. I think I'd rather move it as a class constant on the Service class, since it's much more related to that. But even if it's moved there it's not exactly macOS/Linux dependent. It's init system dependent.

Reply:
Yeh, point taken on caveats, could be a new file. Init system is effectively mapped to OS, though, because we're not supporting systemd on macOs or launchd on Linux.

Reply:
But there might be Linux distros with <insert other option here>. Things like `init.d` files, still exist in the world, arguably even more likely for Linuxbrew users.

Reply:
@SMillerDev yeh but I guess my point is: systemd logic will only ever be applicable on Linux and launchd logic on macOS so it makes sense to scope them into their relative OS layers.

Reply:
Still have a mild preference for this being done.